tianshou/env/worker/subproc.py

```diff
@@ -86,25 +87,40 @@ def _encode_obs(
                 p.close()
                 break
             if cmd == "step":
-                if data is None:  # reset
-                    obs = env.reset()
+                obs, reward, done, info = env.step(data)
+                if obs_bufs is not None:
+                    _encode_obs(obs, obs_bufs)
+                    obs = None
+                p.send((obs, reward, done, info))
+            elif cmd == "reset":
+                retval = env.reset(**data)
+                if reset_returns_info is None:
```

Comment:
This seems a little ad-hoc. Can we do this:

- If a tuple, encode the first element as obs, and keep the rest as is.
- Otherwise, encode the whole object as obs.

Reply:
addressed in https://github.com/thu-ml/tianshou/pull/613/commits/edfcbcf3017d71132f235c0a5a4681811d7eeb6b. Earlier it was suggested to cache the result of `reset_returns_info`, but due to other difficulties we didn't end up doing it in other places.